src/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -749,31 +748,39 @@ private DynamicMetaObject CallMethodNoResult<TBinder>(string methodName, TBinder
             /// </summary>
             private bool IsOverridden(string method)
             {
-                var methods = Value.GetType().GetMember(method, BindingFlags.Public | BindingFlags.Instance).OfType<MethodInfo>();
+                MemberInfo[] members = Value.GetType().GetMember(method, BindingFlags.Public | BindingFlags.Instance);
 
-                foreach (MethodInfo mi in methods)
+                foreach (MemberInfo member in members)
```

Comment:
If bullish to avoid the cost of Linq2Objects, maybe also index and avoid the enumerator allocation?